.github/scripts/ext-change-detector.ts

```diff
@@ -0,0 +1,335 @@
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the
+ * repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { simpleGit } from 'simple-git';
+import { readdirSync, existsSync } from 'fs';
+import { join } from 'path';
+import { BuildContext, ChangeDetectionResult, ExtensionInfo } from './types';
+import { log, setOutput, getExtensionInfo } from './utils';
+
+/**
+ * Get all available VS Code extensions (packages with publisher field)
+ */
+function getAvailableExtensions(): ExtensionInfo[] {
+  const extensions: ExtensionInfo[] = [];
+  const packagesDir = join(process.cwd(), 'packages');
+
+  if (!existsSync(packagesDir)) {
+    log.warning('packages directory not found');
+    return extensions;
+  }
+
+  const packageDirs = readdirSync(packagesDir, { withFileTypes: true })
+    .filter((dirent) => dirent.isDirectory())
+    .map((dirent) => dirent.name);
+
+  for (const packageName of packageDirs) {
+    const packagePath = join(packagesDir, packageName);
+    const packageJsonPath = join(packagePath, 'package.json');
+
+    if (existsSync(packageJsonPath)) {
+      try {
+        const info = getExtensionInfo(packagePath);
+
+        // Only include packages that have a publisher (VS Code extensions)
+        if (info.publisher) {
+          extensions.push({
+            name: packageName,
+            path: packagePath,
+            currentVersion: info.version,
+            publisher: info.publisher,
+            displayName: info.displayName,
+          });
+          log.debug(
+            `Found VS Code extension: ${packageName} (publisher: ${info.publisher})`,
+          );
+        } else {
+          log.debug(`Skipping NPM package: ${packageName} (no publisher)`);
+        }
+      } catch (error) {
+        log.warning(`Failed to read package.json for ${packageName}: ${error}`);
+      }
+    }
+  }
+
+  return extensions;
+}
+
+/**
+ * Check if extension has changes since last release
+ */
+async function hasExtensionChanges(
+  git: any,
+  extensionPath: string,
+  lastTag: string | null,
+): Promise<boolean> {
+  if (!lastTag) {
+    // No previous tag, check if extension has any files
+    const files = readdirSync(extensionPath, { recursive: true });
+    return files.length > 0;
+  }
+
+  try {
+    // Check for changes since the last release tag
+    const diff = await git.diff([lastTag, 'HEAD', '--', extensionPath]);
+    return diff.trim().length > 0;
+  } catch (error) {
+    log.warning(`Failed to check changes for ${extensionPath}: ${error}`);
+    return false;
+  }
+}
+
+/**
+ * Find the last release tag
+ */
+async function findLastReleaseTag(git: any): Promise<string | null> {
+  try {
+    const tags = await git.tags();
+    const versionTags = tags.all
+      .filter((tag: string) => tag.startsWith('v'))
+      .sort((a: string, b: string) => {
+        // Simple version comparison (could be improved with semver)
```

Comment:
This seems like it is fine for now, do we want to take the suggestion to implement semver for more robust version handling later to account for cases where we could have non-numeric symbols? Or simply hold the standard of never doing that with git-tag version naming?

Reply:
This might break with the naming suffix -pre-release as it is now in the git-tag generator. ext-version-bumper.ts

Reply:
The version bumps inside pjson files would be in line with vscode M.m.p, the tags will be expressive as to version plus release type.